### PR TITLE
Fix position upload for heatmap

### PR DESCRIPTION
**Pull request by AdrienRem**

### CODE REVIEW

**Review by TheBlueHawk — APPROVED** [review]

**Review by lucieho — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java [hunk 1]
```diff
@@ -42,7 +42,10 @@ void setInteractor(GridFirestoreInteractor interactor) {
     public CompletableFuture<Void> registerLocation(Carrier carrier, Location location, Date time) {
         refreshLastPositions(time, location);
         Map<String, Object> element = new HashMap<>();
-        element.put("geoPoint", new GeoPoint(location.getLatitude(), location.getLongitude()));
+        element.put("geoPoint", new GeoPoint(
+                location.getLatitude()/CachingDataSender.EXPAND_FACTOR,
```
Comment (TheCrazyKing) [inline app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: can't you add a test to see if the getLatitude is not already in [-99,99 ] and do the division only if it is not ? 
In that way it is more safe

**Review by TheCrazyKing — CHANGES_REQUESTED** [review]
Nice idea, check my comment

File: app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java (hunk 1 above)
Comment (AdrienRem) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: I don't think it's a good idea to consider that our schema could change on a whim. It's unlikely that the schema will change in the foreseeable future

Comment (TheCrazyKing) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: ok

Comment (TheCrazyKing) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: but I don't know if every part of our program will use this datasender correctly

Comment (TheCrazyKing) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: Did you talk with Matteo about that ?
I will accept your PR but trust you you will discuss that point before

**Review by TheCrazyKing — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java (hunk 1 above)
Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: You should use the constant COORDINATE_PRECISION from GridFirestoreInteractor

Comment (matteopariset) [reply app/src/main/java/ch/epfl/sdp/contamination/ConcreteCachingDataSender.java:46]: Actually, I think there is a bigger problem here. The conversion from int (as appearing in the grid label on Firestore) to a valid double should happen **before** creating the Location (argument of registerLocation) @TheCrazyKing

**Pull request merged**